docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+
+__TBD__ (Note: Service is not serving data => no CRUD)
+
+**Guidelines**
+- The inherited `getFactory()` can be used to instantiate the underlying classes.
+- The methods MAY throw exceptions.
+- Single-item-flow methods SHOULD be avoided as they are NOT scalable.
+- Multi-item-flow methods SHOULD operate in batches to be scalable.
+- Multi-item-flow `Create` and `Update` methods SHOULD return a list of entities + error list.
+
+<details><summary markdown='span'>Additional Conventions for Project Development</summary>
+- Definition of interface is NOT recommended. <br/>
+- Specification of each method MUST be included in the facade class instead.
+</details>
+
+<details><summary markdown='span'>Additional Conventions for Module Development</summary>
+- All methods MUST add `@api`, and `{@inheritdoc}` tags on their method documentation. <br/>
+- Single-item-flow methods MUST be avoided as they are NOT scalable.
+</details>
+
+## Factory
+
+```
+[Organization]
+├── Zed
+│   └── [Module]
+│       ├── Communication
+│       │   └── [Module]CommunicationFactory.php
+│       │
+│       ├── Business
+│       │   └── [Module]BusinessFactory.php
+│       │
+│       └── Persistence
+│           └── [Module]PersistenceFactory.php
+├── Yves
+│   └── [Module]
+│       └── [Module]Factory.php
+├── Glue
+│   └── [Module]
+│       └── [Module]Factory.php
+├── Client
+│   └── [Module]
+│       └── [Module]Factory.php
+└── Service
+    └── [Module]
+        └── [Module]ServiceFactory.php
+```
+
+**Description**
+
+The responsibility of a `Factory` is to instantiate classes of a layer and inject dependencies during instantiation.
+
+**Conventions**
+- `Factories` MUST NOT define/implement interface as practically they are never fully replaced.
+- `Factories` MUST orchestrate the instantiation of objects in solitude (without reaching out to class-external logic).
+- Factory methods MUST either instantiate one new object and named as `create[Class]()`, or wire the dependencies and named as `get[Class]()`.
+- Factory methods MUST be `public`.
+- Factory methods MUST use the constants defined in [dependency provider](#dependency-provider) when reaching out to module dependencies.
+
+<details><summary markdown='span'>Additional Conventions for Project Development</summary>
+- Factory methods can define/implement interfaces on demand.<br/>
+- Factory methods can instantiate and write dependencies in one step, but this case MUST be named as `get[Class]()`.<br/>
+- Factory methods can have any access modifier.
+</details>
+
+## Gateway Controller
+
+```
+[Organization]
+└── Zed
+    └── [Module]
+        └── Communication
+            └── Controller
+                └── GatewayController.php
+```
+
+**Description**
+
+`Gateway Controller` is special, reserved `Controller` (see [Controllers](#controller)) in SCOS, that serves as an entry point in [Zed](#zed) for serving [Client](#client) requests (see [Zed Stub](#zed-stub) for more details).
```

Comment:
I would also add that Gateway controller are used only from Zed or BackofficeGateway applications and are not accessible in Backoffice one